### PR TITLE
Making tests run all at once, thereby enabling testing in build definition

### PR DESCRIPTION
**Pull request by kartang**

### CODE REVIEW

File: test/EmptyApp.FunctionalTests/Startup.cs [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
-using Microsoft.ApplicationInsights;
-using Microsoft.ApplicationInsights.AspNet.Extensions;
-using Microsoft.ApplicationInsights.Channel;
-using Microsoft.ApplicationInsights.DataContracts;
-using Microsoft.AspNet.Builder;
-using Microsoft.AspNet.Diagnostics;
-using Microsoft.AspNet.Http;
-using Microsoft.Extensions.Configuration;
-using Microsoft.Extensions.DependencyInjection;
+﻿using Xunit;
```
Comment (dnduffy) [inline test/EmptyApp.FunctionalTests/Startup.cs]: Would be nice to keep the usings together.

File: test/Microsoft.ApplicationInsights.AspNet.Tests/Extensions/ApplicationInsightsExtensionsTests.cs [hunk 2]
```diff
@@ -101,14 +102,17 @@ public static void RegistersTelemetryConfigurationFactoryMethodThatReadsInstrume
...
+                string existingActiveIkey = TelemetryConfiguration.Active.InstrumentationKey;
+
                 ServiceCollection services = ApplicationInsightsExtensionsTests.GetServiceCollectionWithContextAccessor();
-                IConfiguration config = new ConfigurationBuilder().AddJsonFile("content\\config-instrumentation-key.json").Build();
+                IConfiguration config = new ConfigurationBuilder().AddJsonFile("content\\config-instrumentation-key-new.json").Build();
 
+                Assert.NotEqual(existingActiveIkey, config.GetSection(InstrumentationKeyFromConfig).Value);
                 services.AddApplicationInsightsTelemetry(config, false);
-
+
```
Comment (dnduffy) [inline test/Microsoft.ApplicationInsights.AspNet.Tests/Extensions/ApplicationInsightsExtensionsTests.cs]: blank space

**Comment by dnduffy** [comment]
:shipit:

File: test/Microsoft.ApplicationInsights.AspNet.Tests/Extensions/ApplicationInsightsExtensionsTests.cs (hunk 2 above)
Comment (kartang) [inline test/Microsoft.ApplicationInsights.AspNet.Tests/Extensions/ApplicationInsightsExtensionsTests.cs]: Removed!

**Pull request merged**